internal/lookoutv2/repository/getjobs.go

```diff
@@ -0,0 +1,376 @@
+	priorityClass      sql.NullString
+	latestRunId        sql.NullString
+}
+
+type runRow struct {
+	jobId       string
+	runId       string
+	cluster     string
+	node        sql.NullString
+	pending     time.Time
+	started     sql.NullTime
+	finished    sql.NullTime
+	jobRunState int
+	exitCode    sql.NullInt32
+}
+
+type annotationRow struct {
+	jobId           string
+	annotationKey   string
+	annotationValue string
+}
+
+func NewSqlGetJobsRepository(db *pgxpool.Pool) *SqlGetJobsRepository {
+	return &SqlGetJobsRepository{
+		db:           db,
+		queryBuilder: &QueryBuilder{lookoutTables: NewTables()},
+	}
+}
+
+func (r *SqlGetJobsRepository) GetJobs(ctx context.Context, filters []*model.Filter, order *model.Order, skip int, take int) (*GetJobsResult, error) {
+	var err error
+	tx, err := r.db.BeginTx(ctx, pgx.TxOptions{
+		IsoLevel:       pgx.RepeatableRead,
+		AccessMode:     pgx.ReadWrite,
+		DeferrableMode: pgx.Deferrable,
+	})
+	if err != nil {
+		log.WithError(err).Error("failed to start transaction")
+		return nil, err
+	}
+	defer func() {
+		if err != nil {
+			log.WithError(err).Error("transaction failed, rolling back")
+			txCloseErr := tx.Rollback(ctx)
+			if txCloseErr != nil {
+				log.WithError(txCloseErr).Error("failed to roll back transaction")
+			}
+		} else {
+			txCloseErr := tx.Commit(ctx)
+			if txCloseErr != nil {
+				log.WithError(txCloseErr).Error("failed to commit transaction")
+			}
+			err = txCloseErr
+		}
+	}()
+
+	countQuery, err := r.queryBuilder.JobCount(filters)
+	if err != nil {
+		return nil, err
+	}
+	rows, err := tx.Query(ctx, countQuery.Sql, countQuery.Args...)
+	if err != nil {
+		return nil, err
+	}
+	count, err := database.ReadInt(rows)
+	if err != nil {
+		return nil, err
+	}
+
+	createTempTableQuery, tempTableName := r.queryBuilder.CreateTempTable()
+	_, err = tx.Exec(ctx, createTempTableQuery.Sql, createTempTableQuery.Args...)
+	if err != nil {
+		return nil, err
+	}
+
+	insertQuery, err := r.queryBuilder.InsertIntoTempTable(tempTableName, filters, order, skip, take)
+	if err != nil {
+		return nil, err
+	}
+	_, err = tx.Exec(ctx, insertQuery.Sql, insertQuery.Args...)
+	if err != nil {
+		return nil, err
+	}
+
+	jobRows, err := makeJobRows(ctx, tx, tempTableName)
+	if err != nil {
+		log.WithError(err).Error("failed getting job rows")
+		return nil, err
+	}
+	runRows, err := makeRunRows(ctx, tx, tempTableName)
+	if err != nil {
+		log.WithError(err).Error("failed getting run rows")
+		return nil, err
+	}
+	annotationRows, err := makeAnnotationRows(ctx, tx, tempTableName)
+	if err != nil {
+		log.WithError(err).Error("failed getting annotation rows")
+		return nil, err
+	}
+
+	jobs, err := rowsToJobs(jobRows, runRows, annotationRows, order)
+	if err != nil {
+		return nil, err
+	}
+	return &GetJobsResult{
+		Jobs:  jobs,
+		Count: count,
+	}, nil
+}
+
+func rowsToJobs(jobRows []*jobRow, runRows []*runRow, annotationRows []*annotationRow, order *model.Order) ([]*model.Job, error) {
```

Comment:
so this probably a dumb question, but why do we have both `model.Job` and the swagger job.  Could we convert to swagger job straight away or does that lose useful infomation due to the limitations of json?